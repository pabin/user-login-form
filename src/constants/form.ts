export const FORM_HEADING = "Let's set up your account";
export const ALREADY_HAVE_ACCOUNT_MSG = "Already have an account?";
export const SIGIN_IN = "Sign in";

// form fields
export const YOUR_NAME = "Your Name";
export const EMAIL_ADDRESS = "Email Address";
export const PASSWORD_LABEL = "Password";
export const BUTTON_TEXT = "Next";
export const SELECT_TITLE = "I would describe my user type as";
export const SELECT_OPTION1 = "Developer";
export const SELECT_OPTION2 = "QA Engineer";
export const SELECT_OPTION3 = "System Admin";

export const EMAIL_VALIDATION_MESSAGE = "Please enter a valid email address";
export const PASSWORD_HELP_TEXT = "Minimum 8 Characters";

export const TERMS_TEXT = 'Terms of Service';
export const PRIVACY_TEXT = 'Privacy Policy';
export const AGGREEMENT_TEXT = `By clicking the "Next" button, you agree to creating a free account, and to`;

// input types
export const EMAIL = 'email';
export const PASSWORD = 'password';
export const TEXT = 'text';

export const BUTTON_LABEL = 'Next';