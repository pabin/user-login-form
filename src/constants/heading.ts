export const HEADING_TITLE = 'Dummy Heading';
export const HEADING_DESCRIPTION = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore at dolore magna aliqua';