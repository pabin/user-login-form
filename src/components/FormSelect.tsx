import React from 'react';
import { SELECT_TITLE, SELECT_OPTION1, SELECT_OPTION2, SELECT_OPTION3 } from '../constants/form';
import styles from './styles/formInput.module.css';

type FormSelectProps = {
  setValue: (value: string) => void,
}

const FormSelect  = ({ setValue }: FormSelectProps) => {
  return (
    <div className={styles.container}>
      <select data-testid="form-select" name="cars" id="cars" className={styles.select} onChange={(event) => setValue(event.target.value)}>
        <option value="">{ SELECT_TITLE }</option>
        <option value={SELECT_OPTION1}>{ SELECT_OPTION1 }</option>
        <option value={SELECT_OPTION2}>{ SELECT_OPTION2 }</option>
        <option value={SELECT_OPTION3}>{ SELECT_OPTION3 }</option>
      </select>
    </div>
  )
}

export default FormSelect;