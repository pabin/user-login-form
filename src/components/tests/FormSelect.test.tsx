import React from 'react';
import { render, fireEvent } from '@testing-library/react'
import FormSelect from '../FormSelect';
import { SELECT_OPTION1, SELECT_OPTION3 } from '../../constants/form';

describe('FormSelect', () => {
  const mockedSetValue = jest.fn();

  test('renders FormSelect component correctly', async () => {
    const wrapper = render(<FormSelect setValue={mockedSetValue} />);
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  test('set Developer when first option is selected', async () => {
    const wrapper = render(<FormSelect setValue={mockedSetValue} />);

    const selectInput = wrapper.getByTestId('form-select');

    fireEvent.change(selectInput, {
      target: { value: SELECT_OPTION1},
    })

    expect(mockedSetValue).toHaveBeenCalledWith(SELECT_OPTION1);
  });

  test('set System Admin when last option is selected', async () => {
    const wrapper = render(<FormSelect setValue={mockedSetValue} />);

    const selectInput = wrapper.getByTestId('form-select');

    fireEvent.change(selectInput, {
      target: { value: SELECT_OPTION3},
    })
    expect(mockedSetValue).toHaveBeenCalledWith(SELECT_OPTION3);
  });
});