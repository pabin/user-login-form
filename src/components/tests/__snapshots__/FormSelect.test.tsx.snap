// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSelect renders FormSelect component correctly 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <select
        class="select"
        data-testid="form-select"
        id="cars"
        name="cars"
      >
        <option
          value=""
        >
          I would describe my user type as
        </option>
        <option
          value="Developer"
        >
          Developer
        </option>
        <option
          value="QA Engineer"
        >
          QA Engineer
        </option>
        <option
          value="System Admin"
        >
          System Admin
        </option>
      </select>
    </div>
  </div>
</body>
`;
